python/mxnet/gluon/data/batchify.py

```diff
@@ -0,0 +1,433 @@
+
+    for i, arr in enumerate(arrs):
+        if arr.shape == ret_shape[1:]:
+            ret[i] = arr
+        else:
+            slices = [slice(None) for _ in range(arr.ndim)]
+            for pad_axis in range(arr.ndim):
+                slices[pad_axis] = slice(0, arr.shape[pad_axis])
+                assert slices[pad_axis].start != slices[pad_axis].stop
+            slices = [slice(i, i + 1)] + slices
+            ret[tuple(slices)] = arr
+
+
+    ctx = Context('cpu_shared', 0) if use_shared_mem else cpu()
+    ret = _arr.array(ret, ctx=ctx, dtype=dtype)
+    original_length = _arr.array(original_length, ctx=ctx, dtype=np.int32)
+
+    return ret, original_length
+
+
+class Pad(object):
+    """Pad the input ndarrays along the specific padding axis and stack them to get the output.
+    Input of the function will be N samples. Each sample should contain a single element that
+    can be 1) numpy.ndarray, 2) mxnet.nd.NDArray, 3) list of numbers.
+    You can set the `pad_val` to determine the padding value.
+
+    The arrays will be padded to the largest dimensions(at most 5 dimensions to pad) and then
+    stacked to form the final output. In addition, the function will output the original dimensions
+    at the `axis` if ret_length is turned on.
+    Parameters
+    ----------
+    pad_val : float or int, default None
+        The padding value.
+    ret_length : bool, default False
+        Whether to return the valid length in the output.
+    dtype : str or numpy.dtype, default None
+        The value type of the output. If it is set to None, the input data type is used.
+    Examples
+    --------
+    >>> from gluoncv.data import batchify
+    >>> # Inputs are multiple lists
+    >>> a = [1, 2, 3, 4]
+    >>> b = [4, 5, 6]
+    >>> c = [8, 2]
+    >>> batchify.Pad()([a, b, c])
+    [[ 1  2  3  4]
+     [ 4  5  6  0]
+     [ 8  2  0  0]]
+    <NDArray 3x4 @cpu(0)>
+    >>> # Also output the lengths
+    >>> a = [1, 2, 3, 4]
+    >>> b = [4, 5, 6]
+    >>> c = [8, 2]
+    >>> batchify.Pad(ret_length=True)([a, b, c])
+    (
+     [[1 2 3 4]
+      [4 5 6 0]
+      [8 2 0 0]]
+     <NDArray 3x4 @cpu(0)>,
+     [4 3 2]
+     <NDArray 3 @cpu(0)>)
+    >>> # Inputs are multiple ndarrays
+    >>> import numpy as np
+    >>> a = np.array([[1, 2, 3, 4], [5, 6, 7, 8]])
+    >>> b = np.array([[5, 8], [1, 2]])
+    >>> batchify.Pad(pad_val=-1)([a, b])
+    [[[ 1  2  3  4]
+      [ 5  6  7  8]]
+     [[ 5  8 -1 -1]
+      [ 1  2 -1 -1]]]
+    <NDArray 2x2x4 @cpu(0)>
+    >>> # Inputs are multiple NDArrays
+    >>> import mxnet as mx
+    >>> a = nd.array([[1, 2, 3, 4], [5, 6, 7, 8]])
+    >>> b = nd.array([[5, 8], [1, 2]])
+    >>> batchify.Pad(pad_val=-1)([a, b])
+    [[[ 1.  2.  3.  4.]
+      [ 5.  6.  7.  8.]]
+     [[ 5.  8. -1. -1.]
+      [ 1.  2. -1. -1.]]]
+    <NDArray 2x2x4 @cpu(0)>
+    """
+    def __init__(self, pad_val=None, ret_length=False, dtype=None, use_shared_mem=False):
```

Comment:
In the new numpy version of GluonNLP, I removed `ret_length` and just call `pad_val` as `val`. Also, I removed the warning. In addition, there is a round_to flag as in https://github.com/dmlc/gluon-nlp/blob/3f7465ab5d0f926c2c6f424644daaa30668570ba/src/gluonnlp/data/batchify/batchify.py#L222